# Part -1 Python Basics (Variables)

1. Print Your Name with your Father name and Date of birth using suitable escape sequence charactor

In [1]:
name = "Abdul Rauf"
father_name = "Ali AKbar"
dob = "15/07/2005"

print("Name: " + name + "\nFather's Name: " + father_name + "\nDate of Birth: " + dob)


Name: Abdul Rauf
Father's Name: Ali AKbar
Date of Birth: 15/07/2005


2. Write your small bio using variables and print it using print function

In [3]:
name = "Abdul Rauf"
age = 19
city = "Karachi"
occupation = "Student"

print("Name: " + name)
print("Age: " + str(age))
print("City: " + city)
print("Occupation: " + occupation)


Name: Abdul Rauf
Age: 19
City: Karachi
Occupation: Student


3. Write a program in which use all the operators we can use in Python

In [5]:

a = 10
b = 5

# Arithmetic operators
print("Addition: ", a + b)
print("Subtraction: ", a - b)
print("Multiplication: ", a * b)
print("Division: ", a / b)
print("Floor Division: ", a // b)
print("Modulus: ", a % b)
print("Exponentiation: ", a ** b)


print("Equality: ", a == b)
print("Not Equal: ", a != b)
print("Greater Than: ", a > b)
print("Less Than: ", a < b)

print("And: ", a > 5 and b < 10)
print("Or: ", a > 5 or b > 10)
print("Not: ", not(a > 5))

c = a
print("Assignment: ", c)

lst = [1, 2, 3, 4]
print("Membership: ", 2 in lst)

print("Identity: ", a is b)


Addition:  15
Subtraction:  5
Multiplication:  50
Division:  2.0
Floor Division:  2
Modulus:  0
Exponentiation:  100000
Equality:  False
Not Equal:  True
Greater Than:  True
Less Than:  False
And:  True
Or:  True
Not:  False
Assignment:  10
Membership:  True
Identity:  False


4. Completes the following steps of small task:
    - Mention Marks of English , Islamiat and Maths out of 100 in 3 different variables
    - Mention Variable of Total Marks and assign 300 to it
    - Calculate Percentage

In [7]:
# Write your code here
english = 80
islamiat = 70
maths = 90

total_marks = 300
percentage = ((english + islamiat + maths) / total_marks) * 100

print("English Marks: ", english)
print("Islamiat Marks: ", islamiat)
print("Maths Marks: ", maths)
print("Total Marks: ", total_marks)
print("Percentage: ", percentage)


English Marks:  80
Islamiat Marks:  70
Maths Marks:  90
Total Marks:  300
Percentage:  80.0


# Part -2 Python Basics (Conditional Statements)

1) A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years.
Ask user for their salary and year of service and print the net bonus amount.

In [9]:
# Type your code here
salary = float(input("Enter your salary: "))
years_of_service = int(input("Enter your years of service: "))

if years_of_service > 5:
    bonus = salary * 0.05
    print(f"Your net bonus is: Rs. {bonus}")
else:
    print("You are not eligible for a bonus.")


Enter your salary:  50000
Enter your years of service:  10


Your net bonus is: Rs. 2500.0



2) Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [11]:
# Type your code here
age = int(input("Enter your age: "))

if age > 17:
    print("You are eligible to vote.")
else:
    print("You are not eligible to vote.")


Enter your age:  19


You are eligible to vote.



3) Write a program to check whether a number entered by user is even or odd.

In [13]:
# Type your code here
num = int(input("Enter a number: "))

if num % 2 == 0:
    print("The number is even.")
else:
    print("The number is odd.")
#Type your code here

Enter a number:  3


The number is odd.


4) Write a program to check whether a number is divisible by 7 or not.
Show Answer

In [15]:
# Type your code here
num = int(input("Enter a number: "))

if num % 7 == 0:
    print("The number is divisible by 7.")
else:
    print("The number is not divisible by 7.")


Enter a number:  7


The number is divisible by 7.


5) Write a program to display 
"Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [17]:
# Type your code here
num = int(input("Enter a number: "))

if num % 5 == 0:
    print("Hello")
else:
    print("Bye")

Enter a number:  25


Hello


7) Write a program to display the last digit of a number.


In [19]:
# Type your code here
length = float(input("Enter the length of the rectangle: "))
breadth = float(input("Enter the breadth of the rectangle: "))

if length == breadth:
    print("It is a square.")
else:
    print("It is a rectangle.")


Enter the length of the rectangle:  12
Enter the breadth of the rectangle:  10


It is a rectangle.


9) Take values of length and breadth of a rectangle from user and print if it is square or rectangle.

In [21]:
# Type your code here
quantity = int(input("Enter the quantity: "))
cost_per_unit = 100
total_cost = quantity * cost_per_unit

if total_cost > 1000:
    total_cost *= 0.9  

print("Total cost is Rs.", total_cost)


Enter the quantity:  3


Total cost is Rs. 300


10) Take two int values from user and print greatest among them.

In [23]:
# Type your code here
num1 = int(input("Enter the first number: "))
num2 = int(input("Enter the second number: "))

if num1 > num2:
    print(f"{num1} is the greatest.")
else:
    print(f"{num2} is the greatest.")


Enter the first number:  10
Enter the second number:  25


25 is the greatest.


11) A shop will give discount of 10% if the cost of purchased quantity is more than 1000.
Ask user for quantity
Suppose, one unit will cost 100.
Judge and print total cost for user.

In [25]:
# Type your code here
quantity = int(input("Enter the quantity: "))
cost_per_unit = 100
total_cost = quantity * cost_per_unit

if total_cost > 1000:
    total_cost *= 0.9  # Applying 10% discount

print("Total cost is Rs.", total_cost)


Enter the quantity:  350


Total cost is Rs. 31500.0


12) A school has following rules for grading system:

a. Below 25 - F

b. 25 to 45 - E

c. 45 to 50 - D

d. 50 to 60 - C

e. 60 to 80 - B

f. Above 80 - A

Ask user to enter marks and print the corresponding grade.

In [27]:
# Type your code here
marks = float(input("Enter your marks: "))

if marks < 25:
    grade = "F"
elif 25 <= marks < 45:
    grade = "E"
elif 45 <= marks < 50:
    grade = "D"
elif 50 <= marks < 60:
    grade = "C"
elif 60 <= marks < 80:
    grade = "B"
else:
    grade = "A"

print("Your grade is:", grade)


Enter your marks:  80


Your grade is: A


14)A student will not be allowed to sit in exam if his/her attendence is less than 75%.

Take following input from user

- Number of classes held

- Number of classes attended.

And print

- percentage of class attended

- Is student is allowed to sit in exam or not.


In [29]:
# Type your code here
classes_held = int(input("Enter the number of classes held: "))
classes_attended = int(input("Enter the number of classes attended: "))

attendance_percentage = (classes_attended / classes_held) * 100
print(f"Attendance percentage: {attendance_percentage}%")

if attendance_percentage >= 75:
    print("You are allowed to sit in the exam.")
else:
    print("You are not allowed to sit in the exam.")


Enter the number of classes held:  12
Enter the number of classes attended:  10


Attendance percentage: 83.33333333333334%
You are allowed to sit in the exam.


15) Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.

In [31]:
# Type your code here
classes_held = int(input("Enter the number of classes held: "))
classes_attended = int(input("Enter the number of classes attended: "))
medical_cause = input("Do you have a medical cause? (Y/N): ")

attendance_percentage = (classes_attended / classes_held) * 100
print(f"Attendance percentage: {attendance_percentage}%")

if attendance_percentage >= 75 or medical_cause == 'Y':
    print("You are allowed to sit in the exam.")
else:
    print("You are not allowed to sit in the exam.")


Enter the number of classes held:  10
Enter the number of classes attended:  3
Do you have a medical cause? (Y/N):  Y


Attendance percentage: 30.0%
You are allowed to sit in the exam.


16) Write a program to check if a year is leap year or not.

If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [33]:
# Type your code here
year = int(input("Enter a year: "))

if (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0):
    print(f"{year} is a leap year.")
else:
    print(f"{year} is not a leap year.")


Enter a year:  300


300 is not a leap year.


17) Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.

if employee is female, then she will work only in urban areas.

if employee is a male and age is in between 20 to 40 then he may work in anywhere

if employee is male and age is in between 40 t0 60 then he will work in urban areas only.

And any other input of age should print "ERROR"

In [35]:
# Type your code here
age = int(input("Enter your age: "))
gender = input("Enter your gender (M/F): ")
marital_status = input("Enter your marital status (Y/N): ")

if gender == 'F':
    print("You will work in urban areas only.")
elif gender == 'M':
    if 20 <= age <= 40:
        print("You may work anywhere.")
    elif 40 < age <= 60:
        print("You will work in urban areas only.")
    else:
        print("ERROR")
else:
    print("ERROR")


Enter your age:  12
Enter your gender (M/F):  F
Enter your marital status (Y/N):  N


You will work in urban areas only.


6) Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria :
     Unit                                                     Price  
uptp 100 units                                             no charge
Next 200 units                                              Rs 5 per unit
After 200 units                                             Rs 10 per unit
(For example if input unit is 350 than total bill amount is Rs.3500
(For example if input unit is 97 than total bill amount is Rs.0
(For example if input unit is 150 than total bill amount is Rs.750

13) Take input of age of 3 people by user and determine oldest and youngest among them.

In [37]:
# Type your code here
age1 = int(input("Enter age of person 1: "))
age2 = int(input("Enter age of person 2: "))
age3 = int(input("Enter age of person 3: "))

oldest = max(age1, age2, age3)
youngest = min(age1, age2, age3)

print("Oldest person age:", oldest)
print("Youngest person age:", youngest)


Enter age of person 1:  13
Enter age of person 2:  10
Enter age of person 3:  1


Oldest person age: 13
Youngest person age: 1
